codegenerator/cli/templates/static/codegen/src/db/Migrations.res

```diff
@@ -82,16 +82,30 @@ let createEnumIfNotExists = (sql, enum: Enums.enumType<_>) => {
 
 module EntityHistory = {
   let createEntityHistoryTableFunctions: unit => promise<unit> = async () => {
-    // Creates a function for copying all entities in a table into the entity_history table
-    // when near the head of the chain
     let _ = await sql->unsafe(`
-      CREATE OR REPLACE FUNCTION copy_table_to_entity_history(
-          source_table_name TEXT,
-          block_timestamp INTEGER,
-          chain_id INTEGER,
-          block_number INTEGER,
-          log_index INTEGER
-      )
+      CREATE OR REPLACE FUNCTION safe_record_to_jsonb(input_record RECORD)
+      RETURNS JSONB
+      LANGUAGE plpgsql
+      AS $$
+      DECLARE
+          json_result JSONB;
+      BEGIN
+          -- Convert the record into a JSONB object, iterate over each field
+          SELECT jsonb_object_agg(d.key, 
+                                  CASE 
+                                      WHEN jsonb_typeof(d.value) = 'number' THEN to_jsonb(d.value::TEXT)
+                                      ELSE d.value
+                                  END)
+          INTO json_result
+          FROM jsonb_each(to_jsonb(input_record)) AS d;
+
+          RETURN json_result;
+      END;
+      $$;
+    `)
```

Comment:
Hey @DZakh, a solution I'm thinking of here is to always encode json number types as string. Then we don't lose any precision during this copy. The only thing I need to adapt is that numbers like float and int should be parsed as either a string or or number. What do you think?

Reply:
This doesn't look optimized at all, but considering that it should run only once, I think this is going to be a huge improvement.